app/controllers/todos_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TodosController < ApplicationController
+  before_action :set_todo, except: [:new, :index, :create]
+
+  def index
+    @complete = Todo.where(completed: true).sort_by &:duedate
+    @incomplete = Todo.where(completed: false).sort_by &:duedate
+  end
+
+  def show
+  end
+
+  def new
+    @todo = Todo.new
+    @todo.duedate = Time.now
+  end
+
+  def create
+    @todo = Todo.new(todo_params)
+    @todo.duedate = Time.new(params[:year], params[:month], params[:day])

```

Comment:
Good idea to use the `Time` class to integrate the params this way, simplifies the implementation of the form in the template.